### PR TITLE
bin/xbps-query: simplify search result printing (with truncation)

### PR DESCRIPTION
**Pull request by Duncaen**
This should also fix a useless error from lgtm

### CODE REVIEW

**Comment by xtraeme** [comment]
cool! what's going with travis?

File: bin/xbps-query/search.c [hunk 1]
```diff
@@ -49,47 +49,34 @@ struct search_data {
...
-			printf("%s %s %s\n", inststr, tmp, desc);
+		len = snprintf(sd->linebuf, sd->maxcols, "[%s] %-*s %s",
+		    xbps_pkgdb_get_pkg(xhp, pkgver) ? "*" : "-",
+		    align, pkgver, desc);
+		/* add ellipsis if the line was truncated */
+		if (len >= sd->maxcols && sd->maxcols > 4) {
+			for (unsigned int j = 0; j < 3; j++)
+				sd->linebuf[sd->maxcols-j-1] = '.';
 		}
+		puts(sd->linebuf);
```
Comment (xtraeme) [inline bin/xbps-query/search.c:79]: rofl this reminded me about tcl puts().

**Comment by Duncaen** [comment]
restarted the travis builds, no idea what happened.

**Comment by Duncaen** [comment]
Failed again for some reason oO.

we could also use unicode ellipsis (…) so we would only have to truncate one character, but that would require to allocate again maxcols+3 to have enough bytes for the unicode character at the end.

**Comment by xtraeme** [comment]
do we have to care for 3 or 4 bytes? just allocate string*2 and be done rofl

**Comment by Duncaen** [comment]
The calculation would again trigger the lgtm warning lol. Probably not worth getting 2 characters more from truncated output anyways.

**Comment by xtraeme** [comment]
We just really need to know string length for output less 5 chars; if result-5 > maxcols; then result = result + "...\n"

**Pull request merged**